components/zeekopenbsm/src/openbsmconsumer.cpp

```diff
@@ -0,0 +1,361 @@
+      if (!status.succeeded()) {
+        return status;
+      }
+      event.header.remote_port = ntohs(tok.tt.sockinet_ex32.port);
+      event.header.local_address = "";
+      event.header.local_port = 0;
+      break;
+    }
+    default:
+      return Status::failure("unhandled event type");
+    }
+    if (tok.tt.sockinet_ex32.family == AF_INET) {
+      event.header.family = AF_INET;
+    } else if (tok.tt.sockinet_ex32.family == 26) {
+      event.header.family = 10;
+    } else {
+      event.header.family = 0;
+    }
+    break;
+  }
+  }
+  return Status::success();
+}
+
+Status OpenbsmConsumer::populateEventFromTokens(
+    Event &event, const std::vector<tokenstr_t> &tokens) {
+
+  Status status = Status::success();
+  for (const auto &tok : tokens) {
+    switch (tok.id) {
+    case AUT_HEADER32:
+    case AUT_HEADER64:
+    case AUT_HEADER32_EX:
+    case AUT_HEADER64_EX:
+      status = extractHeader(event, tok);
+      break;
+    case AUT_SUBJECT32:
+    case AUT_SUBJECT64:
+    case AUT_SUBJECT32_EX:
+    case AUT_SUBJECT64_EX:
+      status = extractSubject(event, tok);
+      break;
+    case AUT_RETURN32:
+    case AUT_RETURN64:
+      status = extractReturn(event, tok);
+      break;
+    case AUT_SOCKINET32:
+    case AUT_SOCKINET128:
+      status = extractSocketInet(event, tok);
+      break;
+    }
+    if (!status.succeeded()) {
+      return status;
+    }
+  }
+  return status;
+}
+
+void OpenbsmConsumer::parseRecordIntoTokens() {
+
+  u_char *buffer = nullptr;
+
+  int record_len = au_read_rec(audit_pipe, &buffer);
+
+  if (record_len <= 0) {
+    d->logger.logMessage(IZeekLogger::Severity::Error,
+                         "Failed to read openbsm message: " +
+                             std::string(strerror(errno)));
+    return;
+  }
+
+  tokenstr_t tok;
+  std::vector<tokenstr_t> tokens{};
+  std::optional<u_int16_t> event_id;
+
+  auto bytesread = 0;
+  while (bytesread < record_len) {
+    if (au_fetch_tok(&tok, buffer + bytesread, record_len - bytesread) == -1) {
+      d->logger.logMessage(IZeekLogger::Severity::Error,
+                           "Failed to fetch token from the given buffer: " +
+                               std::string(strerror(errno)));
+      break;
+    }
+    tokens.push_back(tok);
+    switch (tok.id) {
+    case AUT_HEADER32:
+      event_id = tok.tt.hdr32.e_type;
+      break;
+    case AUT_HEADER32_EX:
+      event_id = tok.tt.hdr32_ex.e_type;
+      break;
+    case AUT_HEADER64:
+      event_id = tok.tt.hdr64.e_type;
+      break;
+    case AUT_HEADER64_EX:
+      event_id = tok.tt.hdr64_ex.e_type;
+      break;
+    case AUT_SOCKUNIX:
+      // if current record of tokens contains unix socket then just return
+      return;
+    }
+    bytesread += tok.len;
+  }
+
+  assert(event_id);
+
+  if (subscribed_event_ids.find(event_id.value()) ==
+      subscribed_event_ids.end()) {
+    // we will not store current record in the event_list
```

Comment:
Let's document _why_ instead of _how_, e.g.,
```suggestion
    // Skip events not subscribed to.
```